File: Palindrome/Palindrome.cpp
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

bool isPalindrome(string str) {
    transform(str.begin(), str.end(), str.begin(), ::tolower);
    str.erase(remove(str.begin(), str.end(), ' '), str.end());
    
    string rev = str;
    reverse(rev.begin(), rev.end());
    return str == rev;
}

int main() {
    cout << boolalpha;
    cout << isPalindrome("madam") << endl;    // true
    cout << isPalindrome("racecar") << endl;  // true
    cout << isPalindrome("hello") << endl;    // false
    return 0;
}

File: Palindrome/Palindrome.java
public class Palindrome {
    public static boolean isPalindrome(String str) {
        str = str.toLowerCase().replaceAll("\\s+", "");
        String reversed = new StringBuilder(str).reverse().toString();
        return str.equals(reversed);
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome("madam"));    // true
        System.out.println(isPalindrome("racecar"));  // true
        System.out.println(isPalindrome("hello"));    // false
    }
}
